### PR TITLE
add description and examples for `enum` keyword

### PR DESCRIPTION
**Pull request by AgniveshChaubey**
Initial PR.

Juan, please consider providing some suggestions on the type of examples I should cover, specifically in terms of complexity.

_Edit: This PR addresses #86_

### CODE REVIEW

**Comment by jviotti** [comment]
> Juan, please consider providing some suggestions on the type of examples I should cover, specifically in terms of complexity.

Yeah, I think what I'll do is create GitHub milestones for each vocabulary, and try to put down in advance a list of potentially interesting examples we should cover

File: content/2020-12/validation/enum.markdown [hunk 1]
```diff
@@ -18,3 +18,59 @@ related:
   - vocabulary: applicator
     keyword: oneOf
 ---
+
+ The `enum` keyword specifies a validation constraint for an instance, defining a set of permissible values. The value of the `enum` keyword must be an array containing at least one element, and these elements should be unique. The validation succeeds if the value of the instance matches one of the elements in the `enum` array.
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:22]: Reads well, but an extra space here?

```suggestion
The `enum` keyword specifies a validation constraint for an instance, defining a set of permissible values. The value of the `enum` keyword must be an array containing at least one element, and these elements should be unique. The validation succeeds if the value of the instance matches one of the elements in the `enum` array.
```

File: content/2020-12/validation/enum.markdown [hunk 2]
```diff
@@ -18,3 +18,59 @@ related:
...
+
+## Examples
+
+{{<schema `Schema with string enum`>}}
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "string",
+  "enum": ["red", "green", "blue"]
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:30]: For consistency, can we add spaces after/before square brackets in all the examples?

```suggestion
  "enum": [ "red", "green", "blue" ]
```

File: content/2020-12/validation/enum.markdown [hunk 3]
```diff
@@ -18,3 +18,59 @@ related:
...
+
+{{<instance-fail `Instance with value not present in the enum is invalid`>}}
+"black"
+{{</instance-fail>}}
+
+{{<schema `Schema with number enum`>}}
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "number",
+  "enum": [2, 46, 100]
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:46]:
```suggestion
  "enum": [ 2, 46, 100 ]
```

File: content/2020-12/validation/enum.markdown [hunk 4]
```diff
@@ -18,3 +18,59 @@ related:
...
 ---
+
+ The `enum` keyword specifies a validation constraint for an instance, defining a set of permissible values. The value of the `enum` keyword must be an array containing at least one element, and these elements should be unique. The validation succeeds if the value of the instance matches one of the elements in the `enum` array.
+
+## Examples
+
+{{<schema `Schema with string enum`>}}
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "string",
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:29]: If using `enum`, it is considered a good practices to *not* include `type`, as it is redundant. Can we remove it from the below examples too, and maybe even clarify this point?

File: content/2020-12/validation/enum.markdown [hunk 5]
```diff
@@ -18,3 +18,59 @@ related:
...
+{{</instance-fail>}}
+
+{{<instance-fail `Instance with value having different datatype is invalid`>}}
+"2"
+{{</instance-fail>}}
+
+{{<schema `Schema with mixed-type enum`>}}
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "enum": [ "red", 123, true, { "foo": "bar" }, [ 1, 2 ], null ]
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:65]: Awesome example :+1:

File: content/2020-12/validation/enum.markdown [hunk 6]
```diff
@@ -18,3 +18,59 @@ related:
...
+{{</schema>}}
+
+{{<instance-pass `Instance with value present in the enum is valid`>}}
+true
+{{</instance-pass>}}
+
+{{<instance-fail `Instance with value not present in the enum is invalid`>}}
+{ "foo": "baz" }
+{{</instance-fail>}}
+-  _Without specifying a type, you can utilize enum to accept values of various types._
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:76]: As pointed out above, it is actually considered an anti-pattern to use `enum` + `type`. In these cases, we should not promote `type` at all

**Review by jviotti — CHANGES_REQUESTED** [review]
Minor comments, but generally looks pretty good

File: content/2020-12/validation/enum.markdown (hunk 2 above)
Comment (AgniveshChaubey) [reply content/2020-12/validation/enum.markdown]: sure!

File: content/2020-12/validation/enum.markdown [hunk 7]
```diff
@@ -18,3 +18,59 @@ related:
...
+{{</instance-pass>}}
+
+{{<instance-fail `Instance with value not present in the enum is invalid`>}}
+"black"
+{{</instance-fail>}}
+
+{{<schema `Schema with number enum`>}}
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+"enum": [ 2, 46, 100 ]
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:46]: Indentation issues here?

File: content/2020-12/validation/enum.markdown [hunk 8]
```diff
@@ -18,3 +18,59 @@ related:
   - vocabulary: applicator
     keyword: oneOf
 ---
+
+The `enum` keyword specifies a validation constraint for an instance, defining a set of permissible values. The value of the `enum` keyword must be an array containing at least one element, and these elements should be unique. The validation succeeds if the value of the instance matches one of the elements in the `enum` array.
+
+_**Note:** When using  `enum`, it's recommended not to include `type` as it is redundant. While it's not mandatory to exclude `type` with `enum`, combining `enum` and `type` is considered an anti-pattern, so it's better to avoid doing so._
```
Comment (jviotti) [inline content/2020-12/validation/enum.markdown:24]: I think we can simplify this paragraph by making it a lot more direct:

```suggestion
_**Note:** Using the `type` keyword along the `enum` keyword is considered an anti-pattern, as `enum` constraints instances tighter than `type`._
```

**Review by jviotti — APPROVED** [review]
Two minor final comments, but overall looks good. Feel free to merge when the other two comments are addressed

File: content/2020-12/validation/enum.markdown (hunk 7 above)
Comment (AgniveshChaubey) [reply content/2020-12/validation/enum.markdown]: oops! fixed.

**Comment by AgniveshChaubey** [comment]
Woohoo! First PR merged. 🎉

**Pull request merged**